index.d.ts

```diff
@@ -0,0 +1,1353 @@
+     */
+    UnknownDevice = 12,
+  }
+
+  /**
+   * Android specific error codes.
+   * @name BleAndroidErrorCode
+   */
+  export const enum BleAndroidErrorCode {
+    /**
+     * The device has insufficient resources to complete the intended operation.
+     */
+    NoResources = 0x80,
+    /**
+     * Internal error occurred which may happen due to implementation error in BLE stack.
+     */
+    InternalError = 0x81,
+    /**
+     * BLE stack implementation entered illegal state and operation couldn't complete.
+     */
+    WrongState = 0x82,
+    /**
+     * BLE stack didn't allocate sufficient memory buffer for internal caches.
+     */
+    DbFull = 0x83,
+    /**
+     * Maximum number of pending operations was exceeded.
+     */
+    Busy = 0x84,
+    /**
+     * Generic BLE stack error.
+     */
+    Error = 0x85,
+    /**
+     * Command is already queued up in GATT.
+     */
+    CmdStarted = 0x86,
+    /**
+     * Illegal parameter was passed to internal BLE stack function.
+     */
+    IllegalParameter = 0x87,
+    /**
+     * Operation is pending.
+     */
+    Pending = 0x88,
+    /**
+     * Authorization failed before performing read or write operation.
+     */
+    AuthFail = 0x89,
+    /**
+     * More cache entries were loaded then expected.
+     */
+    More = 0x8a,
+    /**
+     * Invalid configuration
+     */
+    InvalidCfg = 0x8b,
+    /**
+     * GATT service already started.
+     */
+    ServiceStarted = 0x8c,
+    /**
+     * GATT link is encrypted but prone to man in the middle attacks.
+     */
+    EncrypedNoMitm = 0x8d,
+    /**
+     * GATT link is not encrypted.
+     */
+    NotEncrypted = 0x8e,
+    /**
+     * ATT command was sent but channel is congested.
+     */
+    Congested = 0x8f,
+  }
+
+  export interface NativeBleError {
+    errorCode: BleErrorCode;
+    attErrorCode?: BleATTErrorCode;
+    iosErrorCode?: BleIOSErrorCode;
+    androidErrorCode?: BleAndroidErrorCode;
+    reason?: string;
+
+    deviceID?: string;
+    serviceUUID?: string;
+    characteristicUUID?: string;
+    descriptorUUID?: string;
+    internalMessage?: string;
+  }
+
+  export interface BleErrorCodeMessageMapping {
+    [key in BleErrorCode]: string;
+  }
+
+  export class BleError extends Error {
+    /**
+     * Platform independent error code. Possible values are defined in {@link BleErrorCode}.
+     */
+    errorCode: BleErrorCode;
+    /**
+     * Platform independent error code related to ATT errors.
+     */
+    attErrorCode?: BleATTErrorCode;
+    /**
+     * iOS specific error code (if not an ATT error).
+     */
+    iosErrorCode?: BleATTErrorCode;
+    /**
+     * Android specific error code (if not an ATT error).
+     */
+    androidErrorCode?: BleAndroidErrorCode;
+    /**
+     * Platform specific error message.
+     */
+    reason?: string;
+
+    constructor(nativeBleError: NativeBleError | string, errorMessageMapping: BleErrorCodeMessageMapping);
+  }
+
+  export class Device {
```

Comment:
Please declare in the same way as `class Service`:
* `implements NativeDevice`
* Add methods
* All properties are always present (remove `?`) and add `null` to the type if `?` was present.